### PR TITLE
Document the Contour integration with Knative.

### PR DESCRIPTION
**Pull request by mattmoor**

### CODE REVIEW

**Comment by mattmoor** [comment]
Looks like the markdown link failure is from something unrelated to my change, so I'm going to ignore it since that leg isn't required.

File: docs/install/README.md [hunk 1]
```diff
@@ -24,6 +24,7 @@ Knative Services.
 
 Currently, three options exist which provide this functionality:
```
Comment (samodell) [inline docs/install/README.md]:
```suggestion
Currently, four options exist which provide this functionality:
```

File: docs/install/Knative-with-Contour.md [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+First, let's install Knative to manage our serverless applications.
+
+The following commands install the Knative Serving components:
+
+1.  To install Knative, first install the CRDs by running the following `kubectl apply`
+    command. This prevents race conditions during the install, which cause intermittent errors:
+
+        kubectl apply --filename https://github.com/knative/serving/releases/download/{{< version >}}/serving-crds.yaml
+
+2.  To complete the install of Knative and its dependencies, next run the
```
Comment (samodell) [inline docs/install/Knative-with-Contour.md]:
```suggestion
1.  To complete the install of Knative and its dependencies, next run the
```

Auto rendering for numbered lists works now, so let's use that.

File: docs/install/Knative-with-Contour.md [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+    following `kubectl apply` command:
+
+        kubectl apply --filename https://github.com/knative/serving/releases/download/{{< version >}}/serving-core.yaml
+
+3.  Monitor the `knative-serving` namespaces and wait until all of the pods come up with a
```
Comment (samodell) [inline docs/install/Knative-with-Contour.md]:
```suggestion
1.  Monitor the `knative-serving` namespaces and wait until all of the pods come up with a
```

File: docs/install/Knative-with-Contour.md [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+    `STATUS` of `Running`:
+
+    ```
+    kubectl get pods -w --all-namespaces
+    ```
+
+### Installing Contour for Knative
+
+This next step will install both Contour (in the namespace `projectcontour`) and an additional
```
Comment (samodell) [inline docs/install/Knative-with-Contour.md]: Replace line 44 and 45 with these lines:

Contour is installed in the namespace `projectcontour` with an additional controller in `knative-serving` to bridge the two.

To install Contour, enter the following command:

File: docs/install/Knative-with-Contour.md [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+controller in `knative-serving` to bridge the two:
+
+        kubectl apply --filename https://raw.githubusercontent.com/knative/serving/{{< version >}}/third_party/contour-latest/contour.yaml
+
+
+You can then configure Knative Serving to use this integration by adding the following key to
```
Comment (samodell) [inline docs/install/Knative-with-Contour.md]:
```suggestion
To configure Knative Serving to use Contour, add the following key to
```

File: docs/install/Knative-with-Contour.md [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: config-network
+  namespace: knative-serving
+data:
+  ingress.class: contour.ingress.networking.knative.dev
+```
+
+This key can be added by running the command:
```
Comment (samodell) [inline docs/install/Knative-with-Contour.md]:
```suggestion
Enter the following command to add the key:
```

File: docs/install/Knative-with-Contour.md [hunk 7]
```diff
@@ -0,0 +1,174 @@
...
+
+
+## Configuring DNS
+
+Knative dispatches to different services based on their hostname, so it greatly
+simplifies things to have DNS properly configured. For this, we must look up the
+external IP address that Contour received. This can be done with the following command:
+
+```
+$ kubectl get svc -nprojectcontour
```
Comment (samodell) [inline docs/install/Knative-with-Contour.md]:
```suggestion
$ kubectl get svc -n projectcontour
```

File: docs/install/Knative-with-Contour.md [hunk 8]
```diff
@@ -0,0 +1,174 @@
...
+contour-internal   ClusterIP      10.3.9.254    <none>          8001/TCP                     29d
+envoy-external     LoadBalancer   10.3.12.166   35.235.99.116   80:31093/TCP,443:31196/TCP   29d
+envoy-internal     ClusterIP      10.3.11.69    <none>          80/TCP                       29d
+```
+
+
+This external IP can be used with your DNS provider with a wildcard `A` record;
+however, for a basic functioning DNS setup (not suitable for production!) this
+external IP address can be used with `xip.io` in the `config-domain` ConfigMap
+in `knative-serving`. You can edit this with the following command:
```
Comment (samodell) [inline docs/install/Knative-with-Contour.md]:
```suggestion
in `knative-serving`. Enter the following command to edit the ConfigMap:
```

File: docs/install/Knative-with-Contour.md [hunk 9]
```diff
@@ -0,0 +1,174 @@
...
+data:
+  # xip.io is a "magic" DNS provider, which resolves all DNS lookups for:
+  # *.{ip}.xip.io to {ip}.
+  35.235.99.116.xip.io: ""
+```
+
+
+## Running Knative apps
+
+Now that your cluster has Contour & Knative installed, you can run serverless applications with Knative.
```
Comment (samodell) [inline docs/install/Knative-with-Contour.md]:
```suggestion
Now that your cluster has Contour and Knative installed, you can run serverless applications with Knative.
```

File: docs/install/Knative-with-Contour.md [hunk 10]
```diff
@@ -0,0 +1,174 @@
...
+
+Let's deploy an app to test that everything is set up correctly:
+
+
+1. Next, create a `Knative Service`
```
Comment (samodell) [inline docs/install/Knative-with-Contour.md]:
```suggestion
1. Create a `Knative Service`.
```

File: docs/install/Knative-with-Contour.md [hunk 11]
```diff
@@ -0,0 +1,174 @@
...
+             env:
+               - name: TARGET
+                 value: Go Sample v1
+   ```
+
+   ```
+   kubectl apply -f helloworld-go.yaml
+   ```
+
+2. Send a request
```
Comment (samodell) [inline docs/install/Knative-with-Contour.md]:
```suggestion
1. Send a request
```

File: docs/install/Knative-with-Contour.md [hunk 12]
```diff
@@ -0,0 +1,174 @@
...
+
+   You can send a request to the `helloworld-go` service with curl using the `URL` given above:
+
+   ```
+   $ curl http://helloworld-go.default.35.235.99.116.xip.io
+
+   Hello Go Sample v1!
+   ```
+
+Congratulations! You have successfully installed Knative with Contour to manage and route to serverless applications!
```
Comment (samodell) [inline docs/install/Knative-with-Contour.md]:
```suggestion
You have successfully installed Knative with Contour to manage and route to serverless applications!
```

**Pull request merged**